test/e2e/cloud-provider-azure.go

```diff
@@ -93,6 +97,21 @@ func EnsureAzureDiskCSIDriverHelmChart(ctx context.Context, input clusterctl.App
 		}
 		InstallHelmChart(ctx, clusterProxy, kubesystem, azureDiskCSIDriverHelmRepoURL, azureDiskCSIDriverChartName, azureDiskCSIDriverHelmReleaseName, options, "")
 	} else {
+		cluster := &clusterv1.Cluster{}
+		Eventually(func(g Gomega) {
+			g.Expect(mgmtClient.Get(ctx, types.NamespacedName{
+				Namespace: input.Namespace,
+				Name:      input.ClusterName,
+			}, cluster)).To(Succeed())
+			if cluster.Labels != nil {
+				cluster.Labels["azuredisk-csi"] = "true"
+			} else {
+				cluster.Labels = map[string]string{
+					"azuredisk-csi": "true",
+				}
+			}
+			g.Expect(mgmtClient.Update(ctx, cluster)).To(Succeed())
+		}, e2eConfig.GetIntervals(specName, "wait-deployment")...).Should(Succeed())
```

Comment:
nit: can we move the log line to the top of the else before the cluster label update